pkg/accountmanager/client.go

```diff
@@ -11,48 +11,43 @@ import (
 	"github.com/aws/aws-sdk-go/aws/session"
 )
 
+//go:generate mockery -name clienter
 type clienter interface {
-	Config(roleArn *arn.ARN) *aws.Config
+	Config(roleArn *arn.ARN, roleSessionName string) *aws.Config
 	IAM(roleArn *arn.ARN) iamiface.IAMAPI
 }
 
+// Default IAM session name for operations performed by the DCE system
+const systemSessionName = "DCESystem"
+
 // Client helps with client management testing and abstraction
 type client struct {
 	session *session.Session
 	sts     stsiface.STSAPI
-	configs map[string]*aws.Config
 }
 
 // Config configures caching of credentials
-func (c *client) Config(roleArn *arn.ARN) *aws.Config {
+func (c *client) Config(roleArn *arn.ARN, roleSessionName string) *aws.Config {
 
 	// return no config for nil inputs
 	if roleArn == nil {
 		return nil
 	}
 
-	key := roleArn.String()
-
-	// check for cached config
-	if c.configs != nil && c.configs[key] != nil {
-		return c.configs[key]
-	}
-
 	// new creds
-	creds := stscreds.NewCredentialsWithClient(c.sts, roleArn.String())
+	creds := stscreds.NewCredentialsWithClient(c.sts, roleArn.String(), func(p *stscreds.AssumeRoleProvider) {
```

Comment:
have you tested if the returned credentials have a duration you are planning on?

Reply:
@kddejong -- I totally get what you're throwing at me now.

https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/stscreds/#NewCredentialsWithClient

>  The credentials will expire every 15 minutes and the role will be named after a nanosecond timestamp of this operation.

which is fine in golang world, where we're using the `Credentials.Retrieve()` method to auto-refresh as needed. But if we're handing those creds as static strings to end users, this is not going to work.